### PR TITLE
Shipping zone description

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because it adds description to shipping zone.

Backend: https://github.com/mirumee/saleor/pull/6653

Related ticket: [SALEOR-1017](https://app.clickup.com/t/2549495/SALEOR-1017)

**PR intended to be tested with API branch:** SALEOR-1017-shipping-rate-description

### Screenshots

image

### Pull Request Checklist

1. [X] All visible strings are translated with proper context.
1. [X] All data-formatting is locale-aware (dates, numbers, and so on).
1. [X] Translated strings are extracted.
1. [X] Number of API calls is optimized.
1. [X] The changes are tested.
1. [X] Data-test are added for new elements.
1. [X] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1017 As a User I want to add description to the shipping rate](https://app.clickup.com/t/2549495/SALEOR-1017)

File: src/shipping/components/ShippingZoneDetailsPage/ShippingZoneDetailsPage.tsx [hunk 1]
```diff
@@ -33,9 +35,27 @@ import ShippingZoneWarehouses from "../ShippingZoneWarehouses";
 
 export interface FormData extends MetadataFormData {
   name: string;
+  description: string;
   warehouses: string[];
 }
 
+const maxDescriptionLength = 300;
```
Comment (mmarkusik) [inline src/shipping/components/ShippingZoneDetailsPage/ShippingZoneDetailsPage.tsx:42]: Nothing big, but maybe when making consts outside of the component function it'd be easier to tell apart from local component consts if the const was like so MAX_DESCRIPTION_LENGTH?

**Review by mmarkusik — APPROVED** [review]

File: src/shipping/components/ShippingZoneDetailsPage/ShippingZoneDetailsPage.tsx [hunk 2]
```diff
@@ -202,6 +225,41 @@ const ShippingZoneDetailsPage: React.FC<ShippingZoneDetailsPageProps> = ({
...
+                error={description.length > maxDescriptionLength}
+                name={"description"}
+                label={
+                  <div className={classes.labelContainer}>
+                    <div className={classes.label}>
+                      <FormattedMessage defaultMessage="Description" />
+                    </div>
+                    {description?.length > 0 && (
+                      <span>
+                        <FormattedMessage
```
Comment (MarekChoinski) [inline src/shipping/components/ShippingZoneDetailsPage/ShippingZoneDetailsPage.tsx:238]: Nothing big, but in this file there are a lot of `defaultMessage`. We can move all messages into `const messages = defineMessages({...})`

**Review by MarekChoinski — APPROVED** [review]

**Review by MarekChoinski — APPROVED** [review]

**Review by jwm0 — APPROVED** [review]

**Review by jwm0 — CHANGES_REQUESTED** [review]
I'd suggest putting this field inside "General Information" section, the input outside the section looks out of the place and I couldn't find any designs that place it there.

**Review by jwm0 — APPROVED** [review]

**Pull request merged**